### PR TITLE
Hand-process troublesome attributes to fit pmcs conversion tooling

### PR DESCRIPTION
**Pull request by chamons**
- There are a number of constructs that were uncommon enough / difficult to handle
  that were changed by hand / vim macro:
   - Multiple platforms |'ed into one Availability attributes. This was trivial to handle
     in the preprocessor but given the limited number of bits in the shadow enum would be
     difficult to handle in the generator.
   - 32-bit arch Availability attributes were really uncommon and hand processing allowed
     them to be skipped completely
   - A few naming changes smooted transitions between old and new attribute

### CODE REVIEW

File: src/appkit.cs [hunk 1]
```diff
@@ -21865,12 +21865,12 @@ partial interface NSOutlineView : NSAccessibilityOutline {
 		void RemoveItems (NSIndexSet indexes, [NullAllowed] NSObject parent, NSTableViewAnimationOptions animationOptions);
 
 		// - (void)insertRowsAtIndexes:(NSIndexSet *)indexes withAnimation:(NSTableViewAnimationOptions)animationOptions UNAVAILABLE_ATTRIBUTE;
-		[Availability (Unavailable = Platform.iOS_Version | Platform.Mac_Version)]
+		[Availability (Unavailable = Platform.Mac_Version)]
```
Comment (chamons) [inline src/appkit.cs]: This is giving me an apidiff change locally on the main PR but I don't understand what's up.

File: src/appkit.cs [hunk 2]
```diff
@@ -21865,12 +21865,12 @@ partial interface NSOutlineView : NSAccessibilityOutline {
...
 		[Export ("insertRowsAtIndexes:withAnimation:")]
 		void InsertRows (NSIndexSet indexes, NSTableViewAnimationOptions animationOptions);
 
 		// - (void)removeRowsAtIndexes:(NSIndexSet *)indexes withAnimation:(NSTableViewAnimationOptions)animationOptions UNAVAILABLE_ATTRIBUTE;
-		[Availability (Unavailable = Platform.iOS_Version | Platform.Mac_Version)]
```
Comment (chamons) [inline src/appkit.cs]: Same

File: src/avfoundation.cs [hunk 3]
```diff
@@ -8308,7 +8320,7 @@ interface AVCaptureConnection {
 
 		[Since (5,0)]
 		[Export ("videoMaxFrameDuration")]
-		[Availability (Introduced = Platform.iOS_5_0 | Platform.Mac_10_7, Deprecated = Platform.iOS_7_0 /* Only deprecated on iOS */)]
+		[Availability (Introduced = Platform.Mac_10_7, Deprecated = Platform.iOS_7_0 /* Only deprecated on iOS */)]
```
Comment (chamons) [inline src/avfoundation.cs]: Wrong

**Comment by chamons** [comment]
This is part of https://github.com/xamarin/xamarin-macios/pull/3096 split off into it's own PR.

It was done auto-magically via overgrown vim macros and a lot of keystrokes, so has the possibility of having mistakes (@spouliot caught a few already and I just found one).

**Comment by chamons** [comment]
Fixing build along w\ more attributes. Looks like cherry pick wasn't as clean as I thought.

File: src/appkit.cs (hunk 1 above)
Comment (timrisi) [reply src/appkit.cs]: What diff is it showing?

File: src/AudioUnit/AudioUnit.cs [hunk 4]
```diff
@@ -1272,12 +1272,15 @@ enum AudioUnitPropertyIDType { // UInt32 AudioUnitPropertyID
 		ParameterHistoryInfo = 53,
 		Nickname = 54,
 		OfflineRender = 37,
-		[Availability (Introduced = Platform.iOS_8_0 |Platform.Mac_10_3)]
+		[Availability (Introduced = Platform.iOS_8_0)]
+		[Availability (Introduced = Platform.Mac_10_3)]
```
Comment (timrisi) [inline src/AudioUnit/AudioUnit.cs]: Do we even NEED the 10_3 tags?  Don't we only support from like 10.7+ anyway?  Is it worth just dropping these if we're already going to have diffs here?

File: src/foundation.cs [hunk 5]
```diff
@@ -12879,7 +12892,8 @@ interface NSTextCheckingResult : NSSecureCoding, NSCopying {
 		double TimeInterval { get; }
 
 		[Export ("components")]
-		[Availability (Introduced = Platform.iOS_4_0 | Platform.Mac_10_7)]
+		[Availability (Introduced = Platform.iOS_4_0)]
```
Comment (timrisi) [inline src/foundation.cs]: Same question about availability.  IIRC we don't support all the way back to iOS 4.0 anyway, do we need to keep the tags around?

Comment (spouliot) [reply src/foundation.cs]: yes, we can drop < 6.0

File: src/SceneKit/Defs.cs [hunk 6]
```diff
@@ -18,15 +18,17 @@
 namespace XamCore.SceneKit {
 
 	[Watch (3,0)]
-	[Availability (Platform.Mac_10_8 | Platform.iOS_8_0)]
+	[Availability (Platform.Mac_10_8)]
+	[Availability (Platform.iOS_8_0)]
```
Comment (spouliot) [inline src/SceneKit/Defs.cs]: ^ should not those be converted ?

Comment (chamons) [reply src/SceneKit/Defs.cs]: That's a good question. Let me take a look...

**Comment by chamons** [comment]
I'll convert Since as well while I'm at it.

File: src/SceneKit/Defs.cs (hunk 6 above)
Comment (chamons) [reply src/SceneKit/Defs.cs]: Oh, I looked for ```Availability (Introduced =``` but not the common case, and only in the original patch.

**Comment by spouliot** [comment]
The previous, complete build failed some introspection tests - that's likely because some attributes changed.

**Comment by chamons** [comment]
Yeah, now that I think i'm mostly done w\ attribute changes I'm going to see what i broke.

**Comment by spouliot** [comment]
The xtro failure can be ignored. It is due to a sort issue (fixed in PR3110) where a defect can be reported in either WebKit or WatchKit.

**Comment by chamons** [comment]
🎉

File: src/healthkit.cs [hunk 7]
```diff
@@ -2043,17 +2045,20 @@ interface HKWorkoutEvent : NSSecureCoding, NSCopying {
...
 		HKMetadata Metadata { get; }
 
-		[Availability (Deprecated = Platform.iOS_11_0 | Platform.Watch_4_0, Message = "Use 'Create (HKWorkoutEventType, NSDateInterval, HKMetadata)' instead.")]
+		[Deprecated (PlatformName.iOS, 11, 0, message: "Use 'Create (HKWorkoutEventType, NSDateInterval, HKMetadata)' instead.")]
+		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use 'Create (HKWorkoutEventType, NSDateInterval, HKMetadata)' instead.")]
 		[Static, Export ("workoutEventWithType:date:")]
 		HKWorkoutEvent Create (HKWorkoutEventType type, NSDate date);
 
-		[Availability (Introduced = Platform.iOS_10_0 | Platform.Watch_3_0, Deprecated = Platform.iOS_11_0 | Platform.Watch_4_0, Message = "Use 'Create (HKWorkoutEventType, NSDateInterval, HKMetadata)' instead.")]
+		[iOS (10, 0, message: "Use 'Create (HKWorkoutEventType, NSDateInterval, HKMetadata)' instead.")]
```
Comment (chamons) [inline src/healthkit.cs]: This is wrong. I need to review the entire diff now...

Comment (spouliot) [reply src/healthkit.cs]: ouch :(
might be a good test to add (message is only used when deprecating or obsoleting an API)

Comment (chamons) [reply src/healthkit.cs]: That is exactly how I caught it. I cherry picked this to my "remove PMCS avail" work and got:

```/Users/donblas/Programming/macios/master/xamarin-macios/src/healthkit.cs(2056,16): error CS1739: The best overloaded method match for `MonoTouch.ObjCRuntime.iOSAttribute.iOSAttribute(byte, byte, byte)' does not contain a parameter named `message'
/Users/donblas/Programming/macios/master/xamarin-macios/src/build/ios/compat/core.dll (Location of the symbol related to previous error)```

Comment (spouliot) [reply src/healthkit.cs]: 👍

Comment (chamons) [reply src/healthkit.cs]: I did a full search of the diff, and this appears to be isolated. There were 3 in close and 2 of them had one syntax and one another. My eyes my have glazed over and I hit the wrong macro three times.

This section of conversion was only semi-automated.

I also hit a number of "should be converted but weren't due to not matching my pattern". I'm fixing those as well.

**Comment by chamons** [comment]
I was able to port this to my remove PMCS avail PR branch:

https://github.com/xamarin/xamarin-macios/pull/3096

and build with no "new" errors. There was one strange iOS with a message I had to fix, but that matches what was there before and can be fixed in the 2nd PR.

**Pull request merged**